sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.http.policy.UserAgentPolicy;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * This class represents various options to be set on the client. It is set on the builder at the time of building the
+ * client.
```

Comment:
I think we should leave the second sentence out if we ever want to use this in different ways. It also doesn't add much explanation about the usage of this class.